### PR TITLE
[Bug] Determine failure of destructured import of tokens

### PR DESCRIPTION
**Pull request by TimRoe**
## Description of Change

Determined the following:
- The tokens package named `themes` export works fine generally
   - Works when leveraged in our Storybook testing environment within useTheme hook
   - Works when leveraged directly in flagship
      - Added as useLocalTheme to their `utils/hooks.tsx` file (copy/pasted useTheme from components package)
      - Imported into LinkRow component: overrode the background color and toggled light/dark mode as expected
- Given the above, determined bug purely a problem when nested within the `components` package
- After significant testing, determined the issue was related to our custom `useColorScheme` and `useTheme` hooks being in the same file:
   - Did work splitting `useTheme` to its' own file
   - Didn't work splitting `index.tsx`->`style.tsx` (not related to being in the index file)
   - Didn't work subsequently splitting the hooks to `styles.ts` (not related to being in a .tsx)
   - Did work when moving `useColorScheme` back to `styles.tsx`
- Unclear why the above situation was causing a failure or why splitting files fixed it

Resolution: 
- Instituted a practice of hooks living in their own folder/files moving forward
   - Created the `hooks` folder within `utils` since hooks/utils have overlapping purpose of providing general utility function
   - Each hook has a file of its' own within, named the same as the hook function itself
- Further, to meet general conventions of index files being an indexing of folder contents, split the existing utils `index.tsx` (and renamed to .ts) up to:
   - accessibility.ts
   - style.tsx
   - storybook.tsx and OSfunctions.tsx were already split out
- To maintain order, the hooks are being exported via the appropriate related util files instead of directly by the utils index
   - So the `hooks` folder is a bin for all general hooks (component specific hooks can remain with components) with no additional structure

#### Testing Packages

- [0.24.1-alpha.1](https://www.npmjs.com/package/@department-of-veterans-affairs/mobile-component-library/v/0.24.1-alpha.1)
- [0.24.1-alpha.0](https://www.npmjs.com/package/@department-of-veterans-affairs/mobile-component-library/v/0.24.1-alpha.0)
- [0.23.2-alpha.4](https://www.npmjs.com/package/@department-of-veterans-affairs/mobile-component-library/v/0.23.2-alpha.4)
- [0.23.2-alpha.3](https://www.npmjs.com/package/@department-of-veterans-affairs/mobile-component-library/v/0.23.2-alpha.3)
- [0.23.2-alpha.2](https://www.npmjs.com/package/@department-of-veterans-affairs/mobile-component-library/v/0.23.2-alpha.2)
- [0.23.2-alpha.1](https://www.npmjs.com/package/@department-of-veterans-affairs/mobile-component-library/v/0.23.2-alpha.1)
- [0.23.2-alpha.0](https://www.npmjs.com/package/@department-of-veterans-affairs/mobile-component-library/v/0.23.2-alpha.0)
- [0.23.1-alpha.0](https://www.npmjs.com/package/@department-of-veterans-affairs/mobile-component-library/v/0.23.1-alpha.0)

## Testing

Verified `yarn test` still passes.

Web: 
- Validated light/dark mode toggle still works
- Confirmed the Doc Site link still displays correctly
- Confirmed Alert base color content still displays correctly
- Validated Phone/TTY/Text Links still work to try to perform action in browser
- Validated Segmented Control PressableOpacity still behaves as expected

Android/iOS:
- Validated same scenarios as Web above (except Doc that isn't present in mobile)
- Validated all Links with functional action
- Validated screen reader
   - Snackbar autodismiss after 5 seconds w/o action
   - Snackbar doesn't autodismiss w/ action
   - Snackbar doesn't autodismiss after 5 seconds w/o action and screen reader off

In flagship w/ latest alpha build:
- Validated bug error no longer appears on entry to the app
- Validated all Web/Android/iOS steps above relevant to components currently implemented within develop in the app

.
- [x] Tested on iOS 
- [x] Tested on Android 
- [x] Tested on Web

## PR Checklist
Code reviewer validation:
- General
	- [x] PR is linked to ticket(s)
	- [x] PR has `changelog` label applied if it's to be included in the changelog
	- [x] Acceptance criteria: 
		- All satisfied _or_
		- Documented reason for not being performed _or_
		- Split to separate ticket and ticket is linked by relevant AC(s)
	- [x] Above PR sections adequately filled out
	- [x] If any breaking changes, in accordance with the [pre-1.0.0 versioning guidelines](https://github.com/department-of-veterans-affairs/va-mobile-library#versioning-policy): a CU ticket has been created for the VA Mobile App detailing necessary adjustments with the package version that will be published by this ticket
- Code
	- [x] Tests are included if appropriate (or split to separate ticket)
	- [x] New functions have proper TSDoc annotations

## Publish

If changes warrant a new version [per the versioning guidelines](https://github.com/department-of-veterans-affairs/va-mobile-library#versioning-policy) and the PR is approved and ready to merge:
- [x] Merge `main` into branch
- [x] Merge branch to `main`
- [x] Verify that [Check Component Integrations](https://github.com/department-of-veterans-affairs/va-mobile-library/actions/workflows/check-component-integrations.yml) workflow ran successfully
- [x] [Publish new version](https://github.com/department-of-veterans-affairs/va-mobile-library/actions/workflows/publish.yml) after merging

### CODE REVIEW

File: packages/components/tsconfig.json [hunk 1]
```diff
@@ -27,9 +27,9 @@
     // "moduleDetection": "auto",                        /* Control what method is used to detect module-format JS files. */
 
     /* Modules */
-    "module": "NodeNext",                                /* Specify what module code is generated. */
+    "module": "NodeNext" /* Specify what module code is generated. */,
     // "rootDir": "./",                                  /* Specify the root folder within your source files. */
-    "moduleResolution": "NodeNext",                      /* Specify how TypeScript looks up a file from a given module specifier. */
+    "moduleResolution": "NodeNext" /* Specify how TypeScript looks up a file from a given module specifier. */,
```
Comment (TimRoe) [inline packages/components/tsconfig.json:32]: Autoformatting that I hate, but doesn't seem worth fighting now that autoformat-on-save is working and doing it this way. Functional changes for testing were backed out.

File: packages/components/src/utils/hooks/useColorScheme.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+/** Handles return of color scheme based on platform */
+export function useColorScheme(): ColorSchemeName {
+  // If not web Storybook, set with RN useColorScheme hook
+  if (!process.env.STORYBOOK_WEB) {
+    return RNUseColorScheme()
+  } else {
+    try {
+      const webStorybookColorScheme =
+        require('./useWebStorybookColorScheme').useWebStorybookColorScheme // eslint-disable-line
```
Comment (TimRoe) [inline packages/components/src/utils/hooks/useColorScheme.ts:14]: Only functional change moving the hooks to their own files was updating this to point to the new location and renamed-to-fit-hook-naming-convention of starting with `use`.

File: packages/components/src/utils/storybook.tsx [hunk 3]
```diff
@@ -28,11 +31,16 @@ export const DocLink = ({ name, docUrl }: DocProps) => {
...
+          color: 'blue',
+          textDecorationLine: 'underline',
+          lineHeight: 20,
+        }}
         onPress={() => {
           Linking.openURL(docUrl)
         }}>
-        View guidance for the {name} component on the VA Mobile Documentation Site
+        View guidance for the {name} component on the VA Mobile Documentation
+        Site
```
Comment (TimRoe) [inline packages/components/src/utils/storybook.tsx:43]: Autoformatting

**Review by narin — APPROVED** [review]

**Comment by narin** [comment]
Discussed structure with Tim and agreed to move forward with index/categories split since there'll be more non-hook utils that we add down the line.

**Pull request merged**